### DISCUSSION TITLE
How to disable deletion of specific collection entries

### DISCUSSION THREAD

**Discussion by japgroevemaker** [statamic/cms#6382] [Q&A] [answered]
Hi everyone,

Is there a way disable deletion of specific collection entries? For example, i have a homepage which always needs to be present and must not be deleted.

Thanks in advance!

**Comment by joshuablum** [comment, marked as answer]
Hey @japgroevemaker,

If your collection is structured and set to have a root entry, this one can't be deleted from within the CP. It's only the first one in the collection though.
![image](https://user-images.githubusercontent.com/3824203/181490754-ab867237-de5f-4f74-aeaf-794f8a42c39e.png)
![image](https://user-images.githubusercontent.com/3824203/181490770-50801463-46c4-45b2-af13-be415803ca3d.png)

You can also use roles and permissions to prevent certain users from deleting entries from (specific) collections.

I don't think that there is currently a way to lock an entry and prevent it from getting deleted. You can open an issue for this in [statamic/ideas](https://github.com/statamic/ideas/issues) 👍

Hope this helps :)

**Reply by jesseleite** [reply]
@japgroevemaker @joshuablum You can setup an event listener on the `EntrySaving` event and conditionally `return false;` from your `handle()` method to prevent it from being saved, based on whatever logic you require.
